docs: add a note how to fix minimum release age with GAR's maven (#35025)
renovatebot/renovate: docs/usage/configuration-options.md

!!! warning "Warning for Maven users"
    For `minimumReleaseAge` to work, the Maven source must return reliable `last-modified` headers.

>     <!-- markdownlint-disable MD046 -->

Author: I disabled the rule, because it seems to be false negative. because it treats continuation of admonition as a indented code which it is not intended to be, but at the same time MkDocs documentation on admonitions is not clear if multi paragraph admonitions are even possible https://squidfunk.github.io/mkdocs-material/reference/admonitions/#inline-blocks-inline-end

```
MD046/code-block-style Code block style [Expected: fenced; Actual: indented] https://github.com/DavidAnson/markdownlint/blob/v0.37.4/doc/md046.md
```

>     If your custom maven source registry doesnt support `last-modified` header, like GAR (Google Artifact Registry's Maven implementation, then feasible option is to extend maven source registry url with https://repo1.maven.org/maven2 as the first item, then `currentVersionTimestamp` via `last-modified` will be taken from maven central for public dependencies.

Reviewer: Are you sure this doesn't make a _functional_ difference? Are you sure that it applies so widely? e.g. does it only apply if your maven registry is a full proxy/pull-through of maven central in the first place? Otherwise you could get versions on Maven Central which aren't in your own registry

Author: you are right, it makes functional difference if custom maven registry is not full proxy or pull through. although in that case it doesn't make sense to combine both in the same `match` rule. perhaps its worth it to specify explicitly?

Author: clarified that `[`45370dc` (#35025)](https://github.com/renovatebot/renovate/pull/35025/commits/45370dc182c3330c1b672daf8b58da0353ef5db0)`

Reviewer: If your proxy is only for pull-through then you could leave it out of registryUrls completely, right? You'd only need to keep it in if it's a combination of public Maven Central + private packages.

Author: yes, that's what we have, it will be good to clarify that. I've removed full proxy and left only pull-through

>     If your custom maven source registry doesnt support `last-modified` header, like GAR (Google Artifact Registry's Maven implementation, then feasible option is to extend maven source registry url with https://repo1.maven.org/maven2 as the first item, then `currentVersionTimestamp` via `last-modified` will be taken from maven central for public dependencies.

Reviewer 2: ```suggestion
    If your custom Maven source registry does _not_ support the `last-modified` header, like GAR (Google Artifact Registry's Maven implementation) then you can extend the Maven source registry url with `https://repo1.maven.org/maven2` as the first item. Then the `currentVersionTimestamp` via `last-modified` will be taken from Maven central for public dependencies.
```

!!! warning "Warning for Maven users"
    For `minimumReleaseAge` to work, the Maven source must return reliable `last-modified` headers.

    If your custom maven source registry doesnt support `last-modified` header, like GAR (Google Artifact Registry's Maven implementation, then feasible option is to extend maven source registry url with https://repo1.maven.org/maven2 as the first item, then `currentVersionTimestamp` via `last-modified` will be taken from maven central for public dependencies.
    

>     <!-- markdownlint-disable MD046 -->

Author: I disabled the rule, because it seems to be false negative. because it expects fenced code, and it is, but it thinks its indented
```
MD046/code-block-style Code block style [Expected: fenced; Actual: indented] https://github.com/DavidAnson/markdownlint/blob/v0.37.4/doc/md046.md
```
